spec/integration/associations/many_to_many/multiple_spec.rb

```diff
@@ -0,0 +1,111 @@
+RSpec.describe ROM::SQL::Associations::ManyToMany, helpers: true do
+  include_context 'database setup'
+
+  #
+  # eans.id <=> ean_stats.ean_id
+  # contract_ean_stats.ean_stat_id <=> ean_stats.id
+  # contracts.id <=> contract_ean_stats.contract_id
+  #
+  with_adapters do
+    before do
+      conn.create_table(:eans) do
+        primary_key :id
+        column :name, String
+      end
+
+      conn.create_table(:ean_stats) do
+        primary_key :id
+        column :ean_id, Integer
+      end
+
+      conn.create_table(:contract_ean_stats) do
+        primary_key :id
+        column :contract_id, Integer
+        column :ean_stat_id, Integer
+      end
+
+      conn.create_table(:contracts) do
+        primary_key :id
+        column :title, String
+      end
+
+      conf.relation(:eans) do
+        schema(infer: true) do
+          associations do
+            has_many :contracts, view: :for_eans, override: true
+          end
+        end
+      end
+
+      conf.relation(:ean_stats) do
+        schema(infer: true) do
+          associations do
+            belongs_to :ean
+            has_many :contract_ean_stats
+          end
+        end
+      end
+
+      conf.relation(:contract_ean_stats) do
+        schema(infer: true) do
+          associations do
+            belongs_to :contract
+            belongs_to :ean_stat
+          end
+        end
+      end
+
+      conf.relation(:contracts) do
+        schema(infer: true) do
+          associations do
+            has_many :contract_ean_stats
+            has_many :ean_stats, through: :contract_ean_stats
+          end
+        end
+
+        def for_eans(assoc, eans)
+          join(:contract_ean_stats, contract_id: contracts[:id]).
+            join(:ean_stats, ean_id: eans.pluck(:id)).
```

Comment:
@flash-gordon yeah I had a feeling something ain't right. I'd actually appreciate if you could pull in this branch locally and see if this works. IIRC I tried what you wrote here but it didn't work (query was invalid)